server/app/middlewares/authentication.ts

```diff
@@ -20,7 +20,11 @@ export const authenticateJWT = (req: Request, res: Response, next: NextFunction)
       return res.sendStatus(403);
     }
 
-    if (!decoded.username || !decoded.email || !decoded.id) res.sendStatus(403);
+    if (decoded.role === 'General' && PATHS_FORBIDDEN_TO_GENERAL_ROLE.includes(req.path)) {
+      res.sendStatus(403);
```

Comment:
I feel like this logic shouldnt be in the authenticateJwt middleware. I think role checking should be its own middleware

Reply:
We might have like 5 roles at some point with different logic so i feel like we should decouple that from the authentication of the jwt

Reply:
yes sir